### PR TITLE
[MarkScan] Post-testing updates to ValidateBallotPage

### PR DESCRIPTION
**Pull request by kofi-q**
## Overview

Related to: https://github.com/votingworks/vxsuite/issues/4762

Addressing a couple issues found on the `ValidateBallotPage` during testing:
- Removing the icon from the `My Ballot is Incorrect` button to avoid wrapping in the default size mode.
- Updating the wording of the undervote warnings to match the strings used on the printed paper ballot, for consistency.
  - Leaving the `You may still vote in this contest` wording intact for the pre-print review screen, but updating `Votes remaining in this contest:` to `Number of unused votes:` there as well, since it's semantically similar and avoids unnecessary branching in the code.

## Demo Video or Screenshot

| Before  | After |
| ------------- | ------------- |
| Screenshot 2024-04-18 at 13 32 12 | Screenshot 2024-04-18 at 13 45 47 |
| Screenshot 2024-04-18 at 13 46 57 | Screenshot 2024-04-18 at 13 47 52 |

## Testing Plan
- New and updated unit tests
- Manual test to verify strings and audio

## Checklist

- ~[ ] I have added [logging](https://github.com/votingworks/vxsuite/tree/main/libs/logging) where appropriate to any new user actions, system updates such as file reads or storage writes, or errors introduced.~

- [x] I have added a screenshot and/or video to this PR to demo the change
- ~[ ] I have added the "user_facing_change" label to this PR to automate an announcement in #machine-product-updates~

### CODE REVIEW

File: libs/mark-flow-ui/src/components/review.tsx [hunk 1]
```diff
@@ -53,10 +53,15 @@ function CandidateContestResult({
...
   vote = [],
   election,
+  isInterpretationResult,
 }: CandidateContestResultInterface): JSX.Element {
   const district = getContestDistrict(election, contest);
   const remainingChoices = contest.seats - vote.length;
 
+  const noVotesString = isInterpretationResult
+    ? appStrings.noteBallotContestNoSelection()
+    : appStrings.warningNoVotesForContest();
```
Comment (kshen0) [inline libs/mark-flow-ui/src/components/review.tsx:63]: Should we capitalize "no selection" for consistency?

**Review by kshen0 — COMMENTED** [review]
~3rd and 4th screenshot appear to be the same~

Never mind!

File: apps/mark-scan/frontend/src/pages/validate_ballot_page.tsx [hunk 2]
```diff
@@ -103,6 +102,7 @@ export function ValidateBallotPage(): JSX.Element | null {
           precinctId={precinctId}
           votes={votes}
           selectionsAreEditable={false}
+          isInterpretationResult
```
[the remark this answers is not in the document]
Comment (kofi-q) [reply apps/mark-scan/frontend/src/pages/validate_ballot_page.tsx:105]: Oh thanks, completely missed `selectionsAreEditable` - will propagate that into subcomponents instead

File: libs/mark-flow-ui/src/components/review.tsx (hunk 1 above)
Comment (kofi-q) [reply libs/mark-flow-ui/src/components/review.tsx:63]: Yeah, SGTM

**Review by kshen0 — APPROVED** [review]

**Pull request merged**